src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -127,10 +128,14 @@ public static IEnumerable<DType> GetPrimitiveTypes()
         // This is only used by attachment type. DocumentDataType is used to workaround the issue of having cycles in struct type.
         protected readonly DType _attachmentType;
 
-        internal HashSet<IExternalTabularDataSource> AssociatedDataSources { get; private set; }
+        internal HashSet<IExternalTabularDataSource> AssociatedDataSources { get; }
         internal IExternalOptionSet<int> OptionSetInfo { get; }
         internal IExternalViewInfo ViewInfo { get; }
 
+        // Eventually, all display names should come from this centralized source
+        // We should not be using individual DataSource/OptionSet/View references
+        internal HashSet<DisplayNameProvider> DisplayNameProviders { get; }
```

Comment:
>HashSet [](http://example.com/codeflow?start=17&length=7)

Can this be a single item instead of a hashset?

Reply:
Updated to pair of DisplayNameProvider and flag whether to disable display names